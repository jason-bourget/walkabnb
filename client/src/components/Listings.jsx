import React from 'react';
import Listing from './Listing.jsx';
import GridList from '@material-ui/core/GridList'
import GridListTile from '@material-ui/core/GridListTile';
import logo from '../assets/logo.png';
import withWidth, { isWidthUp } from '@material-ui/core/withWidth';

const Listings = (props) => {

  const getGridListCols = () => {
    if (isWidthUp('xl', props.width)) {
      return 4;
    }
    if (isWidthUp('lg', props.width)) {
      return 3;
    }
    if (isWidthUp('md', props.width)) {
      return 2;
    }
    return 1;
  }

  /* Each listing is represented by a GridListTile,
  which the GridList comprises. */
  if (props.listings.length > 0) {
    return (
      <GridList cols={getGridListCols()} cellHeight='auto' style={{position: 'absolute', top: '75px'}}>
        {props.listings.map((listing, index) => {
          return (
            <GridListTile key={index}>
              <Listing listing={listing} key={index}/>
            </GridListTile>
          )
        })}
      </GridList>
    );
  }
  return (
    <img src={logo} alt='logo' height='200px' style={{margin: 'auto', display: 'block', position: 'relative', top: '100px'}}/>
  )
};

export default withWidth()(Listings);