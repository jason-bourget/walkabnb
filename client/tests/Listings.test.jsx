import React from 'react';
import Listings from '../src/components/Listings.jsx';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

const listings = [
  {
    airbnbId: 13815789,
    city: "denver",
    coordinates: (2) ["39.73175", "-104.94928"],
    createdAt: "2019-05-11T03:34:58.241Z",
    id: 54,
    image: "https://a0.muscache.com/im/pictures/7ca1d55d-9f8a-47f1-9819-b4970df3825e.jpg",
    price: 59,
    rating: "5",
    reviews: 272,
    size: ["6 guests", "3 bedrooms", "5 beds", "1 bath"],
    title: "Cozy Craftsman Bungalow near City Center",
    updatedAt: "2019-05-11T03:34:58.241Z",
    url: "https://www.airbnb.com/rooms/13815789",
    walkscore: 70
  },
  {
    airbnbId: 6883917,
    city: "denver",
    coordinates: (2) ["39.71935", "-105.0365"],
    createdAt: "2019-05-11T03:34:58.241Z",
    id: 53,
    image: "https://a0.muscache.com/im/pictures/87153622/85341615_original.jpg",
    price: 43,
    rating: "5",
    reviews: 239,
    size: ["2 guests", "1 bedroom", "1 bed", "1 private bath"],
    title: "Private room, private bathroom and great backyard!",
    updatedAt: "2019-05-11T03:34:58.241Z",
    url: "https://www.airbnb.com/rooms/6883917",
    walkscore: 68
  }
]

test('renders correctly', () => {
  const component = renderer.create(
    <Listings listings={listings}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Listings component', () => {
  it('should render a Card component', () => {
    const wrapper = shallow(<Listings listings={listings}/>);
    expect(wrapper.children()).toHaveLength(2);
  })
})
