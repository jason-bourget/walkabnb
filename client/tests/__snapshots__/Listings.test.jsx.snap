// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="MuiGridList-root-1"
  style={
    Object {
      "margin": -2,
      "position": "absolute",
      "top": "75px",
    }
  }
>
  <li
    className="MuiGridListTile-root-2"
    style={
      Object {
        "height": 504,
        "padding": 2,
        "width": "33.333333333333336%",
      }
    }
  >
    <div
      className="MuiGridListTile-tile-3"
    >
      <div
        className="MuiPaper-root-10 MuiPaper-elevation1-13 MuiPaper-rounded-11 MuiCard-root-9 Listing-card-6"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="MuiCardHeader-root-37"
        >
          <div
            className="MuiCardHeader-avatar-38"
          >
            <div
              aria-label="Walkscore"
              className="MuiAvatar-root-43 MuiAvatar-colorDefault-44"
              style={
                Object {
                  "backgroundColor": "#2196f3",
                }
              }
            >
              70
            </div>
          </div>
          <div
            className="MuiCardHeader-content-40"
          >
            <span
              className="MuiTypography-root-46 MuiTypography-body2-54 MuiCardHeader-title-41"
            >
              Cozy Craftsman Bungalow near City Center
            </span>
            <span
              className="MuiTypography-root-46 MuiTypography-body2-54 MuiTypography-colorTextSecondary-79 MuiCardHeader-subheader-42"
            >
              denver
            </span>
          </div>
        </div>
        <a
          href="https://www.airbnb.com/rooms/13815789"
          target="_blank"
        >
          <div
            className="MuiCardMedia-root-82 Listing-media-7"
            style={
              Object {
                "backgroundImage": "url(\\"https://a0.muscache.com/im/pictures/7ca1d55d-9f8a-47f1-9819-b4970df3825e.jpg\\")",
              }
            }
            title="Listing Image"
          />
        </a>
        <div
          className="MuiCardContent-root-84"
          style={
            Object {
              "backgroundColor": "#2196f3",
            }
          }
        >
          <div
            className="MuiGrid-container-85 MuiGrid-justify-xs-space-evenly-106"
          >
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                6 guests
              </div>
            </div>
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                3 bedrooms
              </div>
            </div>
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                5 beds
              </div>
            </div>
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                1 bath
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiCardContent-root-84"
          style={
            Object {
              "backgroundColor": "#F8F8F8",
              "paddingBottom": "16px",
            }
          }
        >
          <div
            className="MuiGrid-container-85 MuiGrid-justify-xs-space-evenly-106"
          >
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
              >
                $
                59
                /night
              </div>
            </div>
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
              >
                Rating: 
                5
                /5
              </div>
            </div>
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
              >
                272
                 reviews
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </li>
  <li
    className="MuiGridListTile-root-2"
    style={
      Object {
        "height": 504,
        "padding": 2,
        "width": "33.333333333333336%",
      }
    }
  >
    <div
      className="MuiGridListTile-tile-3"
    >
      <div
        className="MuiPaper-root-10 MuiPaper-elevation1-13 MuiPaper-rounded-11 MuiCard-root-9 Listing-card-6"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="MuiCardHeader-root-37"
        >
          <div
            className="MuiCardHeader-avatar-38"
          >
            <div
              aria-label="Walkscore"
              className="MuiAvatar-root-43 MuiAvatar-colorDefault-44"
              style={
                Object {
                  "backgroundColor": "#2196f3",
                }
              }
            >
              68
            </div>
          </div>
          <div
            className="MuiCardHeader-content-40"
          >
            <span
              className="MuiTypography-root-46 MuiTypography-body2-54 MuiCardHeader-title-41"
            >
              Private room, private bathroom and great backyard!
            </span>
            <span
              className="MuiTypography-root-46 MuiTypography-body2-54 MuiTypography-colorTextSecondary-79 MuiCardHeader-subheader-42"
            >
              denver
            </span>
          </div>
        </div>
        <a
          href="https://www.airbnb.com/rooms/6883917"
          target="_blank"
        >
          <div
            className="MuiCardMedia-root-82 Listing-media-7"
            style={
              Object {
                "backgroundImage": "url(\\"https://a0.muscache.com/im/pictures/87153622/85341615_original.jpg\\")",
              }
            }
            title="Listing Image"
          />
        </a>
        <div
          className="MuiCardContent-root-84"
          style={
            Object {
              "backgroundColor": "#2196f3",
            }
          }
        >
          <div
            className="MuiGrid-container-85 MuiGrid-justify-xs-space-evenly-106"
          >
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                2 guests
              </div>
            </div>
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                1 bedroom
              </div>
            </div>
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                1 bed
              </div>
            </div>
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                1 private bath
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiCardContent-root-84"
          style={
            Object {
              "backgroundColor": "#F8F8F8",
              "paddingBottom": "16px",
            }
          }
        >
          <div
            className="MuiGrid-container-85 MuiGrid-justify-xs-space-evenly-106"
          >
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
              >
                $
                43
                /night
              </div>
            </div>
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
              >
                Rating: 
                5
                /5
              </div>
            </div>
            <div
              className="MuiGrid-item-86"
            >
              <div
                className="MuiTypography-root-46 MuiTypography-subtitle2-65"
              >
                239
                 reviews
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </li>
</ul>
`;
